code/iaas/lb-logic/src/main/java/io/cattle/iaas/lb/api/action/LoadBalancerConfigSetListenersActionHandler.java

```diff
@@ -83,9 +74,7 @@ private void removeOldListenerMaps(LoadBalancerConfig config, List<? extends Lon
         }
 
         for (LoadBalancerConfigListenerMap mapToRemove : mapsToRemove) {
-            objectProcessManager.scheduleProcessInstance(LoadBalancerConstants.PROCESS_LB_CONFIG_LISTENER_MAP_REMOVE,
-                    mapToRemove, null);
+            lbService.removeListenerFromConfig(config, mapToRemove.getLoadBalancerListenerId());
```

Comment:
We do if the method is called from another context, when just the listenerId is know (LoadBalancerConfigRemoveListenerActionHandler).